extensions/amp-analytics/0.1/transport.js

```diff
@@ -118,13 +148,223 @@ export class Transport {
+          if (!opt_processResponse) {
+            return;
+          }
+          opt_processResponse(
+              this.type_,
+              /** @type
+               * {!../../../src/3p-analytics-common.AmpAnalytics3pResponse}
+               */ (response));
+        },
+        true);
+    return frameData;
+  }
+
+  /**
+   * Called when a creative no longer needs its cross-domain iframe (for
+   * instance, because the creative has been removed from the DOM).
+   * Once all creatives using a frame are done with it, the frame can be
+   * destroyed.
+   * @param {!HTMLDocument} ampDoc The AMP document
+   * @param {!Object<string,string>} transportOptions The 'transport' portion
+   * of the amp-analytics config object
+   */
+  static doneUsingCrossDomainIframe(ampDoc, transportOptions) {
+    const frameUrl = transportOptions['iframe'];
+    const frameData = Transport.getFrameData(frameUrl);
+    if (Transport.decrementIframeUsageCount_(frameData)) {
+      // Some other instance is still using it
+      return;
+    }
+    ampDoc.body.removeChild(frameData.frame);
+    frameData.responseMessageUnlisten();
+    delete Transport.crossDomainIframes_[frameUrl];
+  }
+
+  /**
+   * Record that one more creative is using this cross-domain iframe
+   * @param {!FrameData} frameData  The cross-domain iframe and
+   * associated data
+   * @return {number}
+   * @private
+   */
+  static incrementIframeUsageCount_(frameData) {
+    return ++(frameData.usageCount);
+  }
+
+  /**
+   * Record that one fewer creative is using this cross-domain iframe
+   * @param {!FrameData} frameData  The cross-domain iframe and
+   * associated data
+   * @return {number}
+   * @private
+   */
+  static decrementIframeUsageCount_(frameData) {
+    return --(frameData.usageCount);
+  }
+
+  /**
+   * Returns whether a url of a cross-domain frame is already known
+   * @param {!string} frameUrl
+   * @return {!boolean}
+   * @VisibleForTesting
+   */
+  static hasCrossDomainIframe(frameUrl) {
+    return hasOwn(Transport.crossDomainIframes_, frameUrl);
+  }
+
+  /**
+   * Create a unique value to differentiate messages from
+   * this particular creative to the cross-domain iframe
+   * @returns {string}
+   * @private
+   */
+  static createUniqueId_() {
+    return String(++(Transport.nextId_));
+  }
+
+  /**
+   * Sends an Amp Analytics trigger event to a vendor's cross-domain iframe,
+   * or queues the message if the frame is not yet ready to receive messages.
+   * @param {!string} event A string describing the trigger event
+   * @param {!Object<string, string>} transportOptions
+   * @VisibleForTesting
+   */
+  sendRequestUsingCrossDomainIframe(event, transportOptions) {
+    const frameData = Transport.getFrameData(transportOptions['iframe']);
+    dev().assert(frameData, 'Trying to send message to non-existent frame');
+    dev().assert(frameData.queue,
+        'Event queue is missing for ' + this.id_);
+    frameData.queue.enqueue(this.id_, event);
```

Comment:
Also:
1. I discovered a bug in transport.js - see code comment in createCrossDomainIframe(). This ID was the key to fixing the bug, so it is certainly used now.
2. I modified the example code in examples/analytics-3p-remote-frame.html in https://github.com/google/amphtml/pull/20/files to show customers that this value is available to them.